odoo/modules/loading.py

```diff
@@ -265,62 +280,48 @@ def load_module_graph(env, graph, status=None, perform_checks=True,
-                        lines.append(f"{module_name}.access_{xmlid},access_{xmlid},{module_name}.model_{xmlid},base.group_user,1,0,0,0")
+                        lines.append(
+                            f"{module_name}.access_{xmlid},access_{xmlid},{module_name}.model_{xmlid},base.group_user,1,0,0,0")
                     _logger.warning('\n'.join(lines))
 
-        updating = tools.config.options['init'] or tools.config.options['update']
-        test_time = test_queries = 0
-        test_results = None
-        if tools.config.options['test_enable'] and (needs_update or not updating):
-            from odoo.tests import loader  # noqa: PLC0415
-            suite = loader.make_suite([module_name], 'at_install')
-            if suite.countTestCases():
-                if not needs_update:
-                    registry.setup_models(env.cr)
-                # Python tests
-                tests_t0, tests_q0 = time.time(), odoo.sql_db.sql_counter
-                test_results = loader.run_suite(suite)
-                report.update(test_results)
-                test_time = time.time() - tests_t0
-                test_queries = odoo.sql_db.sql_counter - tests_q0
-
-                # tests may have reset the environment
-                module = env['ir.module.module'].browse(module_id)
-
-        if needs_update:
-            processed_modules.append(package.name)
-
-            ver = adapt_version(package.data['version'])
-            # Set new modules and dependencies
-            module.write({'state': 'installed', 'latest_version': ver})
-
-            package.load_state = package.state
-            package.load_version = package.installed_version
-            package.state = 'installed'
-            for kind in ('init', 'demo', 'update'):
-                if hasattr(package, kind):
-                    delattr(package, kind)
-            module.env.flush_all()
-
-        extra_queries = odoo.sql_db.sql_counter - module_extra_query_count - test_queries
-        extras = []
-        if test_queries:
-            extras.append(f'+{test_queries} test')
-        if extra_queries:
-            extras.append(f'+{extra_queries} other')
-        _logger.log(
-            module_log_level, "Module %s loaded in %.2fs%s, %s queries%s",
-            module_name, time.time() - module_t0,
-            f' (incl. {test_time:.2f}s test)' if test_time else '',
-            env.cr.sql_log_count - module_cursor_query_count,
-            f' ({", ".join(extras)})' if extras else ''
-        )
-        if test_results and not test_results.wasSuccessful():
-            _logger.error(
-                "Module %s: %d failures, %d errors of %d tests",
-                module_name, test_results.failures_count, test_results.errors_count,
-                test_results.testsRun
+            updating = tools.config.options['init'] or tools.config.options['update']
+            test_time = test_queries = 0
+            test_results = None
```

Comment:
Please don't indent the code that runs the tests.  They must be executable independently of a module's installation or upgrade.  The only thing that is acceptable in this PR is moving the code changing the module's record state.